### PR TITLE
Move verify agentOS Script to eng/common

### PR DESCRIPTION
**Pull request by chidozieononiwu**

### CODE REVIEW

File: eng/common/pipelines/templates/steps/verify-agent-os.yml [hunk 1]
```diff
@@ -0,0 +1,35 @@
+# Template for all Python Scripts in this repository
+parameters: 
+  OSName: ''
```
Comment (weshaggard) [inline eng/common/pipelines/templates/steps/verify-agent-os.yml:3]: Lets change this to OSVmImage

Comment (chidozieononiwu) [reply eng/common/pipelines/templates/steps/verify-agent-os.yml]: Thanks. Fixed

File: eng/common/pipelines/templates/steps/verify-agent-os.yml [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+        import platform
+
+        os_parameter = "${{ parameters.OSVmImage }}".lower()
+        if os_parameter.startswith('mac') or os_parameter.startswith('darwin'):
+            os_parameter = 'macOS'
+        elif os_parameter.startswith('ubuntu') or os_parameter.startswith('linux'):
+            os_parameter = 'Linux'
+        elif os_parameter.startswith('vs') or os_parameter.startswith('win'):
+            os_parameter = 'Windows'
+
```
Comment (weshaggard) [inline eng/common/pipelines/templates/steps/verify-agent-os.yml:25]: We probably want to fail if OsVmImage is empty.

File: eng/common/pipelines/templates/steps/verify-agent-os.yml [hunk 3]
```diff
@@ -0,0 +1,35 @@
+# Template for all Python Scripts in this repository
+parameters: 
+  OSVmImage: ''
```
Comment (weshaggard) [inline eng/common/pipelines/templates/steps/verify-agent-os.yml:3]: To make this template a little easier to consume lets simply expect to find an OSVmImage variable defined and if it isn't we give an error saying that it should be set. That will allow us to consume this template by having one line:

```
- template: /eng/common/pipelines/templates/steps/verify-agent-os.yml
```

Comment (mikeharder) [reply eng/common/pipelines/templates/steps/verify-agent-os.yml:3]: I really like this idea if it works.

File: eng/common/pipelines/templates/steps/verify-agent-os.yml [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+        os_parameter = "${{ parameters.OSVmImage }}".lower()
+        if os_parameter.startswith('mac') or os_parameter.startswith('darwin'):
+            os_parameter = 'macOS'
+        elif os_parameter.startswith('ubuntu') or os_parameter.startswith('linux'):
+            os_parameter = 'Linux'
+        elif os_parameter.startswith('vs') or os_parameter.startswith('win'):
+            os_parameter = 'Windows'
+        else:
+            raise Exception('Variable OSVmImage is empty or has an unexpected value')
```
Comment (weshaggard) [inline eng/common/pipelines/templates/steps/verify-agent-os.yml:26]: For better debugging you should output the value.

`Variable OSVmImage is empty or has an unexpected value [${{ parameters.OSVmImage }}]`

File: eng/common/pipelines/templates/steps/verify-agent-os.yml [hunk 5]
```diff
@@ -0,0 +1,38 @@
+# Template for all Python Scripts in this repository
+parameters:
```
Comment (mikeharder) [inline eng/common/pipelines/templates/steps/verify-agent-os.yml:2]: @chidozieononiwu: The suggestion to remove this parameter was resolved.  Is this parameter supposed to still be here?

Comment (weshaggard) [reply eng/common/pipelines/templates/steps/verify-agent-os.yml:2]: We have the parameter still but we default it to `$(OSVmImage)` I think that approach works well and gives folks the flexibility to pass in another variable if needed when consuming the template.

Comment (mikeharder) [reply eng/common/pipelines/templates/steps/verify-agent-os.yml:2]: Got it, I didn't notice the default value.

**Comment by weshaggard** [comment]
@chidozieononiwu feel free to merge once you have finished your validations in the language PRs.

**Comment by chidozieononiwu** [comment]
> We should file a follow-up item to remove the old template once we integrate this everywhere. Although thinking that might break folks that are doing hotfix builds so perhaps we have to keep it around, similar to the azure-sdk-build-tools templates.

I wonder about how we can start using the tools repo versioning?

**Comment by weshaggard** [comment]
>I wonder about how we can start using the tools repo versioning?

Yes we definitely do need to start using the versioning but that unfortunately will not help with older branches which don't have the ref but the sooner we can get our tool ref versions into master the sooner this type of problem goes away.

**Pull request merged**